### PR TITLE
Added distance tags to marketplace listings

### PR DESCRIPTION
**Pull request by jonmchu**
image

### CODE REVIEW

**Comment by jonmchu** [comment]
Default settings for lat and long are 
lat = 37.785834
lng = -122.406417
so distances are calculated from here, we can change through xcode i believe

**Comment by jonmchu** [comment]
image
Shows about 9.5 miles for first listing

File: components/listings/ListingItem.js [hunk 1]
```diff
@@ -35,6 +35,8 @@ class ListingItem extends React.Component {
     let tag = null;
     if (this.props.active) {
       tag = <Tag text="Active" color={Colors.blue}/>
+    } else {
+      tag = <Tag text={`${this.props.listing.distance.toFixed(1)} mi`} color={Colors.blue}/>
```
Comment (amillman) [inline components/listings/ListingItem.js]: can we use a different color like light gray maybe

File: components/listings/ListingsScreen.js [hunk 2]
```diff
@@ -51,20 +55,42 @@ class ListingsScreen extends React.Component {
     window.EventBus.off(Events.claimCancelled, this._addListing);
   }
 
-  _getListings = () => {
+  _setListings = (lat, lng) => {
```
Comment (amillman) [inline components/listings/ListingsScreen.js]: can we rename to `_sortListings`?

File: components/listings/ListingsScreen.js [hunk 3]
```diff
@@ -28,6 +31,7 @@ class ListingsScreen extends React.Component {
     super(props);
 
     this.state = {
+      error: null,
```
Comment (amillman) [inline components/listings/ListingsScreen.js]: we dont need an error in the state if were just gonna be displaying the error in an alert

File: components/listings/ListingsScreen.js [hunk 4]
```diff
@@ -51,20 +55,42 @@ class ListingsScreen extends React.Component {
...
+    navigator.geolocation.getCurrentPosition(
+      (position) => {
+        this._setListings(position.coords.latitude, position.coords.longitude);
+      },
+      (error) => {
+        Alert.alert(
+        'Uh oh!',
+        'Please turn on Location Services to allow Replate to determine your location and listings nearby.',
+        )
+        this.setState({ error: error.message });
```
Comment (amillman) [inline components/listings/ListingsScreen.js]: dont need this

File: helpers/DistanceUtils.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class DistanceUtils {
+
+  static getDistanceFromLatLonInKm(lat1,lon1,lat2,lon2) {
+    var R = 6371; // Radius of the earth in km
+    var dLat = DistanceUtils.deg2rad(lat2-lat1);
+    var dLon = DistanceUtils.deg2rad(lon2-lon1);
+    var a = Math.sin(dLat/2) * Math.sin(dLat/2) +
+      Math.cos(DistanceUtils.deg2rad(lat1)) * Math.cos(DistanceUtils.deg2rad(lat2)) * Math.sin(dLon/2) * Math.sin(dLon/2);
```
Comment (amillman) [inline helpers/DistanceUtils.js]: what the fuck is going on here?

Comment (amillman) [reply helpers/DistanceUtils.js]: oh the surface distance

Comment (jonmchu) [reply helpers/DistanceUtils.js]: LOLL

**Pull request merged**